src/extension/PostProblems.ts

```diff
@@ -0,0 +1,343 @@
+import {
+  workspace,
+  Uri,
+  Diagnostic,
+  Position,
+  Range,
+  DiagnosticCollection,
+} from 'vscode'
+import * as vscode from 'vscode'
+import type { ResourceError } from './types'
+
+/**
+ *  post problems from resource_errors.json the the PROBLEMS view of vscode
+ */
+const PostProblems = {
+  diagnosticCollection: [] as DiagnosticCollection[],
+
+  /** public methods: */
+
+  /**
+   * resets all diagnostic collections, should be called in the beginning of a ScriptRunner run
+   */
+  resetDiagnosticCollection(): void {
+    this.diagnosticCollection.forEach(collection => {
+      collection.clear()
+    })
+    this.diagnosticCollection = []
+  },
+
+  /**
+   * Posting errors from 'resource_errors.json' to vscode 'PROBLEMS'
+   * @param confFile relative path to the .conf file of the current run
+   * @param ts the time stamp when the run happened
+   * @returns an empty promise
+   */
+  async postProblems(confFile: string, ts: number): Promise<void> {
+    const resourceErrorsFile = 'resource_errors.json'
+    const found = await this.findFile(resourceErrorsFile)
```

Comment:
Yes, I thought that too. we can change this to not look for resource_errors.json in sub-directories. Then we risk for people to open in VSCode a directory with the entire actual project in a sub-directory. I think this can be fine if we add an error message telling the user that 'resource_errors.json is not found in the source directory' or something like this, so people understand that they need to open their project directory as workspace.

by the way - this also happens with relative paths... if there are 2 files with the same name, and I get from resourse_errors.json only a relative path, it points to the first file with the same name that it fines. I can also enforce the above behaviour on them too.